### ISSUE TITLE
(Code Refactoring) Broken with PyTorch 1.1.0, No Seeding, Unnecessary Statements

### ISSUE THREAD

**Issue by GerardMaggiolino** [samet-akcay/ganomaly#42]
This post will highlight multiple issues, which can be broken down at the authors request. In general, they all fall under small code changes to fix or update broken or missing functionality. 

**General usage breaks with PyTorch version 1.1.0.**
- In data.load_data, attributes train_data and train_labels are used rather than data and targets for loading MNIST and CIFAR10 data sets. These are deprecated, read-only properties. This throws an error when attempting to assign new values to the data sets.  
- In model.Ganomaly.set_input, resize_() is used on the data attribute of an array, causing a runtime error to be thrown. This can be fixed with a torch.no_grad() context manager. 

**Several lines of code are unnecessary and unused elsewhere.** 
- In model.Ganomaly.update_netd, self.label is filled twice and then not used. This is only necessary in update_netg for the BCE loss. Note, this loss is not discussed in the current version of the paper on Arxiv, which may be of interest to the authors. 
- self.err_d_real and self.err_d_fake are identical to self.err_d, and unnecessary. 

**Seeding is nonfunctional.**
- opt.manualseed is not used to seed any aspect of the network instantiation or training procedure. This will require additional functionality. 

I will make a pull request for some of these changes; @samet-akcay, please do let me know if I'm misinformed on any of these issues! I may be misinterpreting some of your work. Thank you for an excellent paper and open sourced implementation as well 🙂.

**Comment by GerardMaggiolino** [comment]
A pull request has been made. I don't believe this removes all unnecessary code, but is a start. 

**As briefly mentioned above, there are some issues with the paper on Arxiv.** These may have been fixed in the ACCV publication, but aren't publicly available. 

1.) Under equation 4, subscripts for the weights become adv, adv, adv, rather than adv, con, enc. This is a small typo. 
2.) In reference to equation 4, on page 10, weights w<sub>bce</sub> and w<sub>rec</sub> are used, despite **not being present in the loss in equation 4**. From the code, w<sub>rec</sub> is synonymous with w<sub>con</sub>, and w<sub>bce</sub> is not mentioned elsewhere in the paper. 

Furthermore, the loss for the generator described in the paper is in conflict with the code. The auto-encoder + encoder submodule does not use the provided loss, but uses L<sub>conv</sub> + L<sub>enc</sub> + a standard GAN loss based on logged discriminator output, **not mentioned in the paper**. The discriminator uses solely adversarial loss, separate from the auto-encoder + encoder submodule. This is not represented in equation 4. 

Overall, the loss used in the paper is at odds with the loss used in this code. Does the final paper in ACCV rectify this - if so, are the final results different? What version of code was used for the SOTA results obtained in the Arxiv publication?

**Comment by samet-akcay** [comment]
@GerardMaggiolino, thanks again for the contribution and feedback for the paper. We've started working on the updated version of the paper, where such typos and other issues will be revised. We could continue to discuss the generator and discriminator objective in #28.
